user-guide/Advanced_Modules/SRM/SRM_Quarantine.md

```diff
@@ -0,0 +1,271 @@
+> but this can be customized: see [using custom scripts](xref:SRM_custom_scripts#configuring-the-booking-manager-app-to-use-custom-scripts)
+
+To resolve the quarantined state, the quarantined usages on the booking can be inspected, and they can either be restored or a new resource can be assigned.
+The script example below will show how to restore resources that were put in quarantine. This approach will only work if the bookings that caused the conflict have been updated so the resources are available again for the quarantined booking.
+
+```csharp
+var helper = new ResourceManagerHelper(engine.SendSLNetSingleResponseMessage);
+
+// All bookings in the quarantined state can be retrieved with a filter.
+// We are only interested in quarantined bookings that end in the future here.
+var filter = ReservationInstanceExposers.IsQuarantined.Equal(true)
+                                        .AND(ReservationInstanceExposers.End.GreaterThan(DateTime.UtcNow));
+var quarantinedBookings = helper.GetReservationInstances(filter);
+
+foreach (var booking in quarantinedBookings)
+{
+    // Create a dictionary of the usages that still exist on the booking and their quarantine reference.
+    var existingUsagesWithQuarantineReference = booking.ResourcesInReservationInstance
+                                                       .Where(r => r.QuarantineReference != Guid.Empty)
+                                                       .ToDictionary(r => r.QuarantineReference);
+    
+    foreach (var quarantinedResource in booking.QuarantinedResources)
+    {
+        var quarantinedUsageDefinition = quarantinedResource.QuarantinedResourceUsage;
+
+        // The quarantinedResource contains a list of triggers which keep track of when and why a usage was moved to quarantine.
+        // The script will log these for the sake of example.
+        var logBuilder = new StringBuilder();
+        logBuilder.AppendLine($"Resource with ID '{quarantinedUsageDefinition.GUID}' was moved to quarantine for the following reasons:");
+        foreach (var oneTrigger in quarantinedResource.QuarantineTriggers)
+        {
+            logBuilder.AppendLine($"\tat {oneTrigger.CreatedAt:dd/mm/yyyy HH:mm.ss} because {oneTrigger}");
+        }
+        engine.GenerateInformation(logBuilder.ToString());
+
+        if (existingUsagesWithQuarantineReference.TryGetValue(quarantinedUsageDefinition.QuarantineReference, out var existingUsage))
+        {
+            // This quarantined usage was split off and some usage is still assigned to the booking.
+            // This usually happens when there is a capacity overflow and some capacity is still assigned to the resource
+            
+            // Merge the existing usage with the old usage to restore the resource usage.
+            // If the goal of the script is to assign a completely new resource to resolve the quarantine state, the 'existingUsage'
```

Comment:
It would be interesting to add such an example. The existing or quarantined usage could be reused and a new resource ID can be assigned.

Reply:
Done